SDKAndroid/src/test/kotlin/app/cybrid/sdkandroid/mock/ExternalWalletsApiMock.kt

```diff
@@ -0,0 +1,95 @@
+        every { AppModule.getClient().createService(ExternalWalletsApi::class.java) } returns mockExternalWalletsApi
+        return mockExternalWalletsApi
+    }
+
+    fun mock_createExternalWallet(response: Response<ExternalWalletBankModel>): ExternalWalletsApi {
+
+        this.init()
+        val mockExternalWalletsApi = mockk<ExternalWalletsApi>()
+        coEvery { mockExternalWalletsApi.createExternalWallet(postExternalWalletBankModel = any()) } returns response
+        coEvery { mockExternalWalletsApi.listExternalWallets(customerGuid = any()) } returns Response.success(
+            ExternalWalletBankModelMock.list(listOf(
+                ExternalWalletBankModelMock.mock()
+            ))
+        )
+        every { AppModule.getClient().createService(ExternalWalletsApi::class.java) } returns mockExternalWalletsApi
+        return mockExternalWalletsApi
+    }
+
+    fun mock_deleteExternalWallet(response: Response<ExternalWalletBankModel>): ExternalWalletsApi {
+
+        this.init()
+        val mockExternalWalletsApi = mockk<ExternalWalletsApi>()
+        coEvery { mockExternalWalletsApi.deleteExternalWallet(externalWalletGuid = any()) } returns response
+        coEvery { mockExternalWalletsApi.listExternalWallets(customerGuid = any()) } returns Response.success(
+            ExternalWalletBankModelMock.list(listOf(
+                ExternalWalletBankModelMock.mock_deleted()
+            ))
+        )
+        every { AppModule.getClient().createService(ExternalWalletsApi::class.java) } returns mockExternalWalletsApi
+        return mockExternalWalletsApi
+    }
+}
+
+object ExternalWalletBankModelMock {
+
+    fun list(objects: List<ExternalWalletBankModel>): ExternalWalletListBankModel {
+        return ExternalWalletListBankModel(
+            total = JavaBigDecimal(1),
+            page = JavaBigDecimal(0),
+            perPage = JavaBigDecimal(10),
+            objects = objects
+        )
+    }
+
+    fun mock(): ExternalWalletBankModel {
+        return ExternalWalletBankModel(
+            guid = "1234",
+            name = "Test",
+            asset = "BTC",
+            environment = ExternalWalletBankModel.Environment.sandbox,
+            bankGuid = "1234",
+            customerGuid = "1234",
+            address = "0x1234",
+            tag = "1234",
+            createdAt = java.time.OffsetDateTime.now(),
+            state = ExternalWalletBankModel.State.completed,
+            failureCode = null
+        )
+    }
+
+    fun mock_deleted(): ExternalWalletBankModel {
+        return ExternalWalletBankModel(
+            guid = "1234",
+            name = "Test",
+            asset = "BTC",
+            environment = ExternalWalletBankModel.Environment.sandbox,
+            bankGuid = "1234",
+            customerGuid = "1234",
+            address = "0x1234",
+            tag = "1234",
+            createdAt = java.time.OffsetDateTime.now(),
+            state = ExternalWalletBankModel.State.deleted,
```

Comment:
There is also the state "deleting," which is the state the wallet goes into while it is being deleted by the backend.